pkg/controllers/reconciler.go

```diff
@@ -175,6 +177,16 @@ func UpgradeImages(image *string, initimage *string) (upgrade bool) {
 
 		if parts := strings.SplitN(*s, ":", 2); len(parts) == 2 && len(parts[0]) > 0 {
 			name, version := parts[0], parts[1]
+
+			envVarValue := os.Getenv(strings.ReplaceAll(strings.ToUpper(filepath.Base(name)), "-", "_") + "_SHA")
+
+			if envVarValue != "" && *s != envVarValue {
```

Comment:
Question: should we set the CR's image to also have the sha and then check that the CR's `image` is equal to the env variable?

Reply:
`image` and `initImage` values become void when the bundle owner has set the env variables. As far as I understand it, this is the preferred model. Bundle owner decides the allowed images and overrides user settings via CRs.

Reply:
Sorry for late reply.! Yeap, thanks for answering in my stead.!